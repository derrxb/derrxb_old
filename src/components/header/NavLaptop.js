import React from 'react';
import { Link } from 'gatsby';
import styled from 'styled-components';
import Media from '../shared/Media';

const NavMenu = styled.div`
  margin-left: auto;
  ${Media.lessThan('laptop')`
    display: none;
  `};
`;

export const NavOption = styled(Link)`
  text-decoration: none;
  color: #444;
  margin-right: 1em;
  &:hover {
    border-bottom: 2px solid #111;
    color: #111;
  }
  ${Media.lessThan('laptop')`
    width: 100%;
    height: 100%;
    display: block;
    padding: 0.5em 1em;
    color: #444;
    font-weight: 700;
    text-decoration: none;
    &:focus,
    &:hover {
      border-bottom: none;
      text-decoration: none;
    }
  `};
`;

const NavLaptop = () => (
  <NavMenu>
    <NavOption to="/">Home</NavOption>
    <NavOption to="/blog">Blog</NavOption>
    <NavOption to="/photography">Photography</NavOption>
    <NavOption to="/book">Book</NavOption>
    <NavOption to="/about">About</NavOption>
  </NavMenu>
);

export default NavLaptop;
