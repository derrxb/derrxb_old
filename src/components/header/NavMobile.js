import React from 'react';
import styled from 'styled-components';
import { Spring, config } from 'react-spring/renderprops.cjs';
import Modal from 'react-modal';
import Media from '../shared/Media';
import { NavOption } from './NavLaptop';

const MenuLauncher = styled.button`
  background: #ff9e00;
  color: black;
  border-radius: 3px;
  font-size: 12px;
  font-weight: 700;
  width: 64px;
  padding: 0.5em;
  outline: none;
  border: none;
  display: none;
  text-align: center;
  &:focus {
    outline: none;
  }
  ${Media.lessThan('laptop')`
    display: block;
  `};
`;

const NavDropdownHeader = styled.li`
  display: flex !important;
  align-items: center;
  background: #2d67a7 !important;
  h4 {
    color: white;
    margin: 0;
    padding: 0.5em 1em;
  }
`;

const NavDropdownWrapper = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
  width: 100%;
  li {
    display: block;
    width: 100%;
    height: 48px;
    margin-left: none;
    padding-left: none;
    margin: 0;
    &:nth-of-type(odd) {
      background: #eee;
    }
  }
`;

const ModalStyle = {
  overlay: {
    zIndex: 5,
    background: 'rgba(14,30,37,.54)',
  },
  content: {
    height: 'fit-content',
    width: 'calc(100% - 80px)',
    padding: 0,
    border: 'none',
    borderRadius: '10px',
  },
};

const NavMobile = () => {
  const [showModal, setShowModal] = React.useState(false);

  return (
    <React.Fragment>
      <MenuLauncher onClick={() => setShowModal(!showModal)}>Menu</MenuLauncher>
      {showModal && (
        <Spring
          from={{ transform: 'scale(0)', opacity: 0 }}
          to={{ transform: 'scale(1)', opacity: 1 }}
          config={config.stiff}
        >
          {({ opacity, transform }) => (
            <Modal
              isOpen={showModal}
              style={{
                overlay: {
                  ...ModalStyle.overlay,
                  opacity,
                },
                content: {
                  ...ModalStyle.content,
                  transform,
                },
              }}
              onRequestClose={() => setShowModal(false)}
              contentLabel="Site Menu"
            >
              <NavDropdownWrapper data-testid="menu-nav-modal">
                <NavDropdownHeader>
                  <h4>Menu</h4>
                </NavDropdownHeader>

                <li>
                  <NavOption to="/blog">Blog</NavOption>
                </li>

                <li>
                  <NavOption to="/book">Book Session</NavOption>
                </li>
              </NavDropdownWrapper>
            </Modal>
          )}
        </Spring>
      )}
    </React.Fragment>
  );
};

export default NavMobile;
