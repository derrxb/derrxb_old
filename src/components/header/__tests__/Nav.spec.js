import React from 'react';
import 'jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent } from 'react-testing-library';
import ReactModal from 'react-modal';
import NavMobile from '../NavMobile';
import NavLaptop from '../NavLaptop';

afterEach(cleanup);

ReactModal.setAppElement(document.createElement('div'));

describe('NavModal Specs', () => {
  test('renders the menu modal on `menu` click', () => {
    // Arrange
    const { getByTestId, getByText } = render(<NavMobile />);

    // Act
    fireEvent.click(getByText(/menu/i));

    // Assert
    expect(getByTestId('menu-nav-modal')).toBeInTheDocument();
  });

  it('renders the modal on > desktop correctly', () => {
    const tree = renderer.create(<NavLaptop />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
