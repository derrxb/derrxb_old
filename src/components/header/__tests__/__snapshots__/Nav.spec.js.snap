// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavModal Specs renders the modal on > desktop correctly 1`] = `
<div
  className="sc-bdVaJa iORtBj"
>
  <a
    className="sc-bwzfXH dVvytM"
    href="/"
  >
    Home
  </a>
  <a
    className="sc-bwzfXH dVvytM"
    href="/blog"
  >
    Blog
  </a>
  <a
    className="sc-bwzfXH dVvytM"
    href="/photography"
  >
    Photography
  </a>
  <a
    className="sc-bwzfXH dVvytM"
    href="/book"
  >
    Book
  </a>
</div>
`;
